app/src/main/java/ch/epfl/qedit/view/edit/EditSettingsActivity.java

```diff
@@ -0,0 +1,117 @@
+        Quiz quiz = (Quiz) bundle.getSerializable(QUIZ_ID);
+
+        if (quiz != null) {
+            setContentView(R.layout.activity_edit_modify_quiz_settings);
+
+            // Initialize the builder with the existing quiz
+            quizBuilder = new Quiz.Builder(quiz);
+
+        } else {
+            setContentView(R.layout.activity_edit_new_quiz_settings);
+
+            // Initialize a new empty QuizBuilder
+            quizBuilder = new Quiz.Builder();
+
+            // Create spinner (language list)
+            languageSelectionSpinner = findViewById(R.id.edit_language_selection);
+
+            // Find app's current language position in languages list
+            String currentLanguage = Locale.getDefault().getLanguage();
+            String[] languageList = getResources().getStringArray(R.array.languages_codes);
+            int positionInLanguageList = Arrays.asList(languageList).indexOf(currentLanguage);
+
+            // Set current language in spinner at startup
+            languageSelectionSpinner.setSelection(positionInLanguageList, false);
+
+            // Set listener
+            languageSelectionSpinner.setOnItemSelectedListener(this);
+        }
+
+        // Set the EditText for the title
+        editTitle = findViewById(R.id.edit_quiz_title);
+
+        String title = stringPool.get(TITLE_ID);
+        // TODO Support old questions that store the strings directly as well
+        editTitle.setText((title == null) ? quiz.getTitle() : title);
+    }
+
+    public void startEditing(View view) {
+        // Update the title in the StringPool and the languageCode
+        stringPool.update(TITLE_ID, editTitle.getText().toString());
+        stringPool.setLanguageCode(languageCode);
+
+        // Launch the EditQuizActivity
+        Intent intent = new Intent(EditSettingsActivity.this, EditQuizActivity.class);
+        Bundle bundle = new Bundle();
+        bundle.putSerializable(QUIZ_BUILDER, quizBuilder);
+        bundle.putSerializable(STRING_POOL, stringPool);
+        intent.putExtras(bundle);
+        startActivity(intent);
+    }
+
+    @Override
+    /* This method runs if the user selects another language */
+    public void onItemSelected(AdapterView parent, View view, int pos, long id) {
+        // Do not run if user has not chosen a language
+        if (!userHasInteracted) {
+            return;
+        }
+
+        // Get language code from the position of the clicked language in the spinner
+        languageCode = getResources().getStringArray(R.array.languages_codes)[pos];
+    }
+
+    @Override
+    public void onNothingSelected(AdapterView parent) {
+        // Not used because there will always be something selected
+    }
+
+    @Override
+    /* This method is needed to apply the desired language at the activity startup */
+    protected void attachBaseContext(Context base) {
+        super.attachBaseContext(LocaleHelper.onAttach(base));
+    }
+}
```

Comment:
from what i understand, this activity will become a dialog in the next sprint. Right?

Reply:
Yes, it will simplify the workflow and will also look nicer. I didn't have enough time to do these changes in this sprint. Additionally we will be able to use `startActivityForResult` for the entire edition by eliminating the settings activity.